FlatHub: bump metainfo.xml on release

ThatKalle opened the pull request:
The `bump_metainfo.py` script prepends the latest version to `org.meshtastic.meshtasticd.metainfo.xml`, _well, any version number to any xml file with a `<releases>` element._

Like so:
```xml
$ cat bin/org.meshtastic.meshtasticd.metainfo.xml
...
<releases>
  <release version="2.6.4">
    <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.4</url>
  </release>
</releases>

$ ./bin/bump_metainfo.py --file bin/org.meshtastic.meshtasticd.metainfo.xml 2.6.4
Version 2.6.4 found without date. Adding date...
$ cat bin/org.meshtastic.meshtasticd.metainfo.xml
...
<releases>
  <release version="2.6.4" date="2025-04-15">
    <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.4</url>
  </release>
</releases>

$ ./bin/bump_metainfo.py --file bin/org.meshtastic.meshtasticd.metainfo.xml 2.6.4
Version 2.6.4 is already present with date, skipping insertion.
$ cat bin/org.meshtastic.meshtasticd.metainfo.xml
...
<releases>
  <release version="2.6.4" date="2025-04-15">
    <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.4</url>
  </release>
</releases>

$ ./bin/bump_metainfo.py --file bin/org.meshtastic.meshtasticd.metainfo.xml 2.6.5
Inserted new release: 2.6.5
$ cat bin/org.meshtastic.meshtasticd.metainfo.xml
...
<releases>
  <release version="2.6.5" date="2025-04-15">
    <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.5</url>
  </release>
  <release version="2.6.4" date="2025-04-15">
    <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.4</url>
  </release>
</releases>
```

This logic is then extended to be used within GitHub Actions on a published release in `release_channels.yml`.
The steps performs the prepending, and creates a Pull Request to add the change back to the repo.
Much in the same way as `version.properties` is handled.


This PR also updates the metadata file to match the expected format, and bumps `version.properties` _as that did not trigger on last release_
And configures renovate to keep an eye on the pip requirements.txt

ThatKalle commented:
Moved to using `defusedxml.ElementTree` as per https://github.com/meshtastic/firmware/actions/runs/14431832385/job/40467622294

Script functionality remains the same.
Adds a `pip` dependency though, added in GHA.

vidplace7 commented:
Thanks for putting up with all my requests (via Discord) @ThatKalle 👍 this is coming along nicely!

vidplace7 commented on bin/bump_metainfo/bump_metainfo.py:
```diff
@@ -0,0 +1,71 @@
+        'date': args.date
+    })
+    url = ET.SubElement(new_release, 'url', {'type': 'details'})
+    url.text = "https://github.com/meshtastic/firmware/releases"
```
```suggestion
    url.text = f"https://github.com/meshtastic/firmware/releases?q=tag%3Av{args.version}"
```
I found a fancy cheat to reference the version by the short-name ❤️ Let's use it!

vidplace7 commented on bin/org.meshtastic.meshtasticd.metainfo.xml:
```diff
@@ -87,8 +87,11 @@
   <releases>
-    <release version="v2.6.4.b89355f" date="2025-04-10">
-      <url type="details">https://github.com/meshtastic/firmware/releases/tag/v2.6.4.b89355f</url>
+    <release version="2.6.5" date="2025-04-09">
+      <url type="details">https://github.com/meshtastic/firmware/releases</url>
+    </release>
+    <release version="2.6.4" date="2025-03-29">
+      <url type="details">https://github.com/meshtastic/firmware/releases</url>
     </release>
```
```suggestion
    <release version="2.6.6">
      <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.6</url>
    </release>
    <release version="2.6.5" date="2025-04-09">
      <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.5</url>
    </release>
    <release version="2.6.4" date="2025-03-29">
      <url type="details">https://github.com/meshtastic/firmware/releases?q=tag%3Av2.6.4</url>
    </release>
```
Add v2.6.6 as well as part of the manual-bump in this PR, and use the short-version URL hack :)

vidplace7 commented:
@thebentern please don't cut a new release until this is merged (I will approve it when it's ready).
This PR includes a manual 2.6.6 bump (since the automation failed), and fixes for version bumps going forward.

Really happy to have @ThatKalle stepping up to help with Linux Release Engineering efforts :heart:

ThatKalle commented:
- I've rejiggered the workflow
  It now only produces 1 PR with all 3 updates 👍
  Also sets default shell to bash and ensures `chomd +x` on the scripts.
- Manually bumped the metainfo.xml to 2.6.6 without a date property.
  Added the ability for the `bump_metainfo.py` script to add the date property if it's missing (should only? be needed for this manual version)
- Added the magical url format, **Nice find!!**

vidplace7 commented on .github/workflows/release_channels.yml:
```diff
@@ -63,29 +66,42 @@ jobs:
         uses: peter-evans/create-pull-request@v7
         with:
-          title: Bump version.properties
+          title: "Bump: Bumps on Published release"
```
```suggestion
          title: "Bump release version"
```
I'd try to keep this short and sweet :) and use the word "version" in the title so it's easier to understand.
